packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -113,6 +113,15 @@ export class ZkEvmProvider implements Provider {
       && this.guardianClient !== undefined;
   }
 
+  private async refreshUserLogin() {
```

Comment:
I think the approach we've taken here only partially solves the issue. When we fixed this in `passportImxProvider`, the approach that we took was to never store a reference to the user. Instead, we would always fetch the user from the `authManager`. When `eth_requestAccounts` is called, we pass the `user` to the `RelayerClient` constructor, so even though we're refreshing `this.user` in `zkEvmProvider`, the `user` that the `RelayerClient` uses would still be out of date.

Maybe a better approach here would be to update the `RelayerClient` & `GuardianClient` constructors to accept the `authManager`, that way they can both retrieve the `user` themselves. This would also be more consistent with the approach we've taken in `passportImxProvider`.

Thoughts?

Reply:
To keep the `isLoggedIn()` check nice and simple, we could just store the `ethAddress` and check that 🤔 especially since `eth_sendTransaction` & `eth_signTypedDataV4` only need a user for that address, and to pass the `accessToken` to the `RelayerClient` & `GuardianClient` (but with the above changes, they should get the `accessToken` by calling the `AuthManager`)

Reply:
Good call, will update the solution to reflect that.